bats/tests/profile/sample.bats

```diff
@@ -32,69 +45,188 @@ start_app() {
+
+    run get_setting .kubernetes.version
+    "${assert}_output" "$PROFILE_LOCKED_KUBERNETES_VERSION" || return
+    refute_output "$PROFILE_DEFAULTS_KUBERNETES_VERSION" || return
 }
 
 @test 'initial factory reset' {
     factory_reset
 }
 
-@test 'start up without profile' {
+@test 'start up with NO profiles' {
     RD_USE_PROFILE=false
-    start_app
+    RD_USE_IMAGE_ALLOW_LIST=false
+    start_application
 }
 
-@test 'verify default settings' {
+@test 'verify there are NO profiles created' {
+    before verify_profiles
+}
+
+@test 'verify default settings were applied' {
     before verify_settings
 }
 
-@test 'factory reset' {
+@test 'factory reset before creating profiles' {
     factory_reset
 }
 
-@test 'create profile' {
+@test 'create profiles' {
     PROFILE_TYPE=$PROFILE_LOCKED
     create_profile
-    add_profile_string containerEngine.name "$PROFILE_CONTAINER_ENGINE"
 
     PROFILE_TYPE=$PROFILE_DEFAULTS
     create_profile
+
+    verify_profiles
+}
+
+@test 'add settings to defaults profiles' {
+    PROFILE_TYPE=$PROFILE_DEFAULTS
     add_profile_bool application.startInBackground "$PROFILE_START_IN_BACKGROUND"
+    add_profile_string containerEngine.name "$PROFILE_CONTAINER_ENGINE"
+    add_profile_string kubernetes.version "$PROFILE_DEFAULTS_KUBERNETES_VERSION"
 }
 
-@test 'start app with new profile' {
+@test 'add settings to locked profile' {
+    PROFILE_TYPE="$PROFILE_LOCKED"
+    add_profile_bool containerEngine.allowedImages.enabled "$PROFILE_USE_IMAGE_ALLOW_LIST"
+    add_profile_list containerEngine.allowedImages.patterns "${PROFILE_IMAGE_PATTERNS[@]}"
+    add_profile_bool application.extensions.allowed.enabled "$PROFILE_USE_EXTENSION_LIST"
+    add_profile_list application.extensions.allowed.list "${PROFILE_EXTENSION_LIST[@]}"
+    add_profile_string kubernetes.version "$PROFILE_LOCKED_KUBERNETES_VERSION"
+}
+
+@test 'start app with new profiles' {
     RD_CONTAINER_ENGINE=""
     start_app
 }
 
-@test 'verify profile settings' {
+@test 'verify profile settings were applied' {
+    verify_settings
+}
+
+@test 'try to install a not-allowed extension' {
+    run rdctl extension install ignatandrei/blockly-automation
+    assert_failure || return
+    run rdctl extension install joycelin79/newman-extension:0.0.5
+    assert_failure || return
+}
+
+@test 'install an allowed extension' {
+    run rdctl extension install joycelin79/newman-extension:0.0.7
+    assert_success || return
+}
+
+@test 'try to change locked fields via rdctl set' {
+    run rdctl set --container-engine.allowed-images.enabled=false
+    assert_failure || return
+
+    run rdctl set --kubernetes.version="1.16.15"
+    assert_failure || return
+}
+
+@test 'try to change locked fields via API' {
+    run rdctl api /v1/settings -X PUT --body "{\"version\": \"$RD_API_VERSION\", \"containerEngine\": {\"allowedImages\": { \"patterns\": [ \"pattern1\" ] }}}"
+    assert_failure || return
```

Comment:
I think you should add an `assert_output --partial ...` here to make sure the failure is due to a locked field, and not something else.